### PR TITLE
saml: purge token after first response and improve setting description

### PR DESCRIPTION
**Pull request by rohityadavcloud**
This improves the description of a saml signature checking global setting, and purges the SAML token upon handling the first SAML response.

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)
- [ ] build/CI
- [ ] test (unit or integration test code)

### Testing

The community QA server is applied with these changes to test the SAML SSO with the `rick` user https://qa.cloudstack.cloud/simulator

### CODE REVIEW

**Review by DaanHoogland — APPROVED** [review]
clgtm

**Comment by DaanHoogland** [comment]
@blueorangutan package

**Comment by rohityadavcloud** [comment]
@blueorangutan package

File: plugins/user-authenticators/saml2/src/main/java/org/apache/cloudstack/api/command/SAML2LoginAPIAuthenticatorCmd.java [hunk 1]
```diff
@@ -228,6 +228,7 @@ public String authenticate(final String command, final Map<String, Object[]> par
                             "Received SAML response for a SSO request that we may not have made or has expired, please try logging in again",
                             params, responseType));
                 }
+                samlAuthManager.purgeToken(token);
```
Comment (rohityadavcloud) [inline plugins/user-authenticators/saml2/src/main/java/org/apache/cloudstack/api/command/SAML2LoginAPIAuthenticatorCmd.java:231]: this isn't very useful for now; but potentially guards against any replays

**Comment by rohityadavcloud** [comment]
@blueorangutan package

**Review by shwstppr — APPROVED** [review]
code lgtm

**Comment by DaanHoogland** [comment]
unit test failure :( , trying locally.

**Comment by weizhouapache** [comment]
> unit test failure :( , trying locally.

the unit test with `ActionEventInterceptorTest` ?
it is happening with all prs againt 4.19

**Comment by rohityadavcloud** [comment]
Yup it’s failure on 4.19 branch, anyone fixing it?

**Comment by DaanHoogland** [comment]
it compiles without issues locally

**Comment by rohityadavcloud** [comment]
@blueorangutan package

**Comment by shwstppr** [comment]
Seems like I didn't wait enough while reopening the PR to kick GH actions. Trying again

**Review by sureshanaparti — APPROVED** [review]
clgtm

**Comment by sureshanaparti** [comment]
@blueorangutan test

**Pull request merged**